includes/documents/class-wcpdf-invoice.php

```diff
@@ -151,13 +145,35 @@ public function get_sequential_number_store() {
 	}
 	
 	public function get_sequential_number_store_name( $date ) {
-		$year       = $date->date_i18n( 'Y' );
-		$store_name = "{$this->slug}_number_{$year}";
-
-		// legacy filter for backward compatibility
-		$store_name = apply_filters( 'wpo_wcpdf_document_sequential_number_store', $store_name, $this );
+		$year            = $date->date_i18n( 'Y' );
+		$store_base_name = apply_filters( 'wpo_wcpdf_document_sequential_number_store', "{$this->slug}_number", $this );
+	
+		// migrate old stores without year suffix
+		$number_store_migrated = get_option( "wpo_wcpdf_number_store_{$store_base_name}_migrated" );
+		if( ! $number_store_migrated ) {
+			$this->maybe_migrate_number_store( $store_base_name );
+		}
+	
+		return apply_filters( "wpo_wcpdf_{$this->slug}_number_store_name", "{$store_base_name}_{$year}", $this );
+	}
 
-		return apply_filters( "wpo_wcpdf_{$this->slug}_number_store_name", $store_name, $this );
+	private function maybe_migrate_number_store( $store_base_name ) {
+		global $wpdb;
+		$now            = new \WC_DateTime( "@".time(), new \DateTimeZone( 'UTC' ) );
+		$year           = $now->date_i18n( 'Y' );
+		$new_table_name = "{$store_base_name}_{$year}";
+
+		if( $wpdb->get_var( "SHOW TABLES LIKE '{$store_base_name}'") == $store_base_name ) {
```

Comment:
This doesn't seem right, does this not need the prefix and the filter?
```php
apply_filters( "wpo_wcpdf_number_store_table_name", "{$wpdb->prefix}wcpdf_{$store_name}", $store_name, $method )
```

Reply:
Sorry, don't what you mean.

Reply:
Yes, sorry.

Reply:
You still didn't apply the filter though
```php
$old_table_name = "{$wpdb->prefix}wcpdf_{$store_base_name}"; 
```
Which will not work correctly for multisite with shared invoice numbers (one of the filters from your tests I assume?)
```php
$old_table_name = apply_filters( "wpo_wcpdf_number_store_table_name", "{$wpdb->prefix}wcpdf_{$store_base_name}", $store_base_name, $method ); 
```
you would need to pass `$method` here for full support of the filter though

Reply:
Do you prefer to pass it from function to function or call it again?